### PR TITLE
Add thread id and spans to correlate all the events coming from one score.

### PR DESCRIPTION
**Pull request by agokarn**
All the logs associated with a single client request will have a unique GUID x-ms-request-id
 
  14 {"timestamp":"2024-10-24T03:11:04.083005Z","level":"INFO","fields":{"message":"new"},"target":"ohttp_server","filename":"ohttp-server/src/main.rs","line_number":401,"span":{"guid":"1c63e97d-f03e-416b-bdb7-5942ed89aa34","name":"x-ms-request-id"},"spans":[],"threadId":"ThreadId(40)"}
15 {"timestamp":"2024-10-24T03:11:04.083104Z","level":"INFO","fields":{"message":"enter"},"target":"ohttp_server","filename":"ohttp-server/src/main.rs","line_number":401,"span":{"guid":"1c63e97d-f03e-416b-bdb7-5942ed89aa34","name":"x-ms-request-id"},"spans":[{"guid":"1c63e97d-f03e-416b-bdb7-5942ed89aa34","name":"x-ms-request-id"}],"threadId":"ThreadId(40)"}
16 {"timestamp":"2024-10-24T03:11:04.083164Z","level":"INFO","fields":{"message":"Received encapsulated score request for target http://127.0.0.1:5001/"},"target":"ohttp_server","filename":"ohttp-server/src/main.rs","line_number":404,"span":{"guid":"1c63e97d-f03e-416b-bdb7-5942ed89aa34","name":"x-ms-request-id"},"spans":[{"guid":"1c63e97d-f03e-416b-bdb7-5942ed89aa34","name":"x-ms-request-id"}],"threadId":"ThreadId(40)"}
17 {"timestamp":"2024-10-24T03:11:04.083217Z","level":"INFO","fields":{"message":"Request headers length = 6"},"target":"ohttp_server","filename":"ohttp-server/src/main.rs","line_number":406,"span":{"guid":"1c63e97d-f03e-416b-bdb7-5942ed89aa34","name":"x-ms-request-id"},"spans":[{"guid":"1c63e97d-f03e-416b-bdb7-5942ed89aa34","name":"x-ms-request-id"}],"threadId":"ThreadId(40)"}
18 {"timestamp":"2024-10-24T03:11:04.083436Z","level":"INFO","fields":{"message":"Fetching MAA token from https://maanosecureboottestyfu.eus.attest.azure.net"},"target":"ohttp_server","filename":"ohttp-server/src/main.rs","line_number":210,"span":{"guid":"1c63e97d-f03e-416b-bdb7-5942ed89aa34","name":"x-ms-request-id"},"spans":[{"guid":"1c63e97d-f03e-416b-bdb7-5942ed89aa34","name":"x-ms-request-id"}],"threadId":"ThreadId(40)"}
19 {"timestamp":"2024-10-24T03:11:04.422488Z","level":"INFO","fields":{"message":"Sending SKR request to https://accconfinferencedebug.confidential-ledger.azure.com/app/key?kid=9"},"target":"ohttp_server","filename":"ohttp-server/src/main.rs","line_number":231,"span":{"guid":"1c63e97d-f03e-416b-bdb7-5942ed89aa34","name":"x-ms-request-id"},"spans":[{"guid":"1c63e97d-f03e-416b-bdb7-5942ed89aa34","name":"x-ms-request-id"}],"threadId":"ThreadId(40)"}

### CODE REVIEW

File: ohttp-server/src/main.rs [hunk 1]
```diff
@@ -397,110 +398,115 @@ async fn score(
...
-    let (ohttp, token) = match load_config(&maa_url, &kms_url, kid).await {
-        Err(e) => {
-            error_with_backtrace!(e);
-            return Ok(warp::http::Response::builder().status(500).body(Body::from(
-                &b"Failed to get or load OHTTP configuration."[..],
-            )));
-        }
-        Ok((config, token)) => match OhttpServer::new(config) {
-            Ok(server) => (server, token),
+    let span = info_span!("x-ms-request-id", guid = %Uuid::new_v4());
```
Comment (kapilvgit) [inline ohttp-server/src/main.rs:401]: Wondering if you considered using #[tracing::instrument(level = "info")]

Comment (agokarn) [reply ohttp-server/src/main.rs:401]: So we do not want it for trace or debug?

Comment (kapilvgit) [reply ohttp-server/src/main.rs:401]: I meant as an alternative for async_move.instrument.await pattern, see https://docs.rs/tracing/latest/tracing/struct.Span.html#in-asynchronous-code.

Comment (agokarn) [reply ohttp-server/src/main.rs:401]: I looked and I cannot find a way to add a new random guid each time. we either would have to pass a new guid as parameter and use that  #[tracing::instrument(fields= request_id)] where request_id is a parameter to the function. Would you prefer that?

Comment (kapilvgit) [reply ohttp-server/src/main.rs:401]: Yes that sounds good.

Comment (agokarn) [reply ohttp-server/src/main.rs:401]: I will have to refactor the code and have a score_internal called from score and have #[tracing::instrument(fields= request_id)] for score_internal. WE will still need to refactor.

Comment (agokarn) [reply ohttp-server/src/main.rs]: fixed

**Review by kapilvgit — APPROVED** [review]

**Pull request merged**